tokio-trace-futures/examples/proxy_server.rs

```diff
@@ -0,0 +1,137 @@
+    println!("Proxying to: {}", server_addr);
+
+    let done = socket
+        .incoming()
+        .map_err(|e| {
+            debug!("error accepting socket; error = {:?}", e)
+        })
+        .for_each(move |client| {
+            let server = TcpStream::connect(&server_addr);
+            match client.peer_addr() {
+                Ok(x) => info!("client connected {}", x), 
+                Err(e) => debug!("error, could not get client info: {}", e)
+            }
+            
+            let amounts = server.and_then(move |server| {
+                // Create separate read/write handles for the TCP clients that we're
+                // proxying data between. Note that typically you'd use
+                // `AsyncRead::split` for this operation, but we want our writer
+                // handles to have a custom implementation of `shutdown` which
+                // actually calls `TcpStream::shutdown` to ensure that EOF is
+                // transmitted properly across the proxied connection.
+                //
+                // As a result, we wrap up our client/server manually in arcs and
+                // use the impls below on our custom `MyTcpStream` type.
+                let client_reader = MyTcpStream(Arc::new(Mutex::new(client)));
+                let client_writer = client_reader.clone();
+                let server_reader = MyTcpStream(Arc::new(Mutex::new(server)));
+                let server_writer = server_reader.clone();
+
+                // Copy the data (in parallel) between the client and the server.
+                // After the copy is done we indicate to the remote side that we've
+                // finished by shutting down the connection.
+                let client_to_server = copy(client_reader, server_writer)
+                    .and_then(|(n, _, server_writer)| {
+                        info!(size = n);
+                        shutdown(server_writer).map(move |_| n)
+                    }).instrument(span!(" client_to_server"));
+
+                let server_to_client = copy(server_reader, client_writer)
+                    .and_then(|(n, _, client_writer)| {
+                        info!(size = n);
+                        shutdown(client_writer).map(move |_| n)
+                    }).instrument(span!(" server_to_client"));
+
+                client_to_server.join(server_to_client)
+            });
+
+            let msg = amounts
+                .map(move |(from_client, from_server)| {
+                    info!(
+                        "client wrote {} bytes and received {} bytes",
+                        from_client, from_server
+                    );
+                })
+                .map_err(|e| {
+                    // Don't panic. Maybe the client just disconnected too soon.
+                    debug!("error: {}", e);
+                }).instrument(span!(" transfer_complete"));
+
+            tokio::spawn(msg);
+
+            Ok(())
+        });
```

Comment:
I didn't instrument that because for some reason that was not being printed out. This is the output after I made the changes  : 
```shell
 INFO proxy_server: client connected client_addr=127.0.0.1:45762
 INFO transfer:client_to_server: proxy_server: size=149
 INFO transfer:server_to_client: proxy_server: size=138
 INFO transfer: proxy_server: client_to_server=149 server_to_client=138
```

Reply:
Ah, that's because no events are actually being recorded inside that span. If this wasn't the case (i.e. if we had enabled events emitted by `tokio` or another dependency, or if we added events in that span) we would see it in the logs as well.